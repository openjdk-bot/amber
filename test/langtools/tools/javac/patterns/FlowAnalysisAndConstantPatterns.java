/*
 * @test /nodynamiccopyright/
 * @summary Stress test flow analysis with constant patterns
 * @compile/fail/ref=FlowAnalysisAndConstantPatterns.out -XDrawDiagnostics FlowAnalysisAndConstantPatterns.java
 */
public class FlowAnalysisAndConstantPatterns {
    public static int getInt(){
        return 43;
    }
    public static void main(String[] args) {
        int i = getInt();
        switch (i) {
            case int j: System.out.println("it's an int!"); break;
            case 42: System.out.println("42!"); break;
        }
    }
}
