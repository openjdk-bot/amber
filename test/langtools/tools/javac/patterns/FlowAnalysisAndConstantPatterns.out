FlowAnalysisAndConstantPatterns.java:14:13: compiler.err.pattern.dominated
1 error
